mapreduce/src/main/java/uk/gov/gchq/palisade/PalisadeInputFormat.java

```diff
@@ -0,0 +1,391 @@
+     *
+     * @param context    the job to configure
+     * @param serialiser the serialiser that can decode the value type this job is processing
+     */
+    public static <V> void setSerialiser(final JobContext context, final Serialiser<Object, V> serialiser) {
+        Objects.requireNonNull(context, "context");
+        Objects.requireNonNull(serialiser, "serialiser");
+        setSerialiser(context.getConfiguration(), serialiser);
+    }
+
+    /**
+     * Sets the {@link Serialiser} for a given configuration. This takes the given serialiser and serialisers
+     * <i>that</i> into the configuration. This allows the Hadoop {@link RecordReader} to create the serialiser inside
+     * the MapReduce job for the de-serialisation step before sending the data to the map task. The serialiser will be
+     * serialised into JSON and the resulting string stored inside the Hadoop configuration object.
+     *
+     * @param conf       the job to configure
+     * @param serialiser the serialiser that can decode the value type this job is processing
+     * @throws NullPointerException for null parameters
+     */
+    public static <K extends Object, V> void setSerialiser(final Configuration conf, final Serialiser<K, V> serialiser) {
+        Objects.requireNonNull(conf, "conf");
+        Objects.requireNonNull(serialiser, "serialiser");
+        conf.set(SERIALISER_CLASSNAME_KEY, serialiser.getClass().getName());
+        conf.set(SERLIALISER_CONFIG_KEY, new String(JSONSerialiser.serialise(serialiser), UTF8));
+    }
+
+    /**
+     * Creates the de-serialiser from the information stored in a given job context. This uses a JSON serialiser to
+     * create the serialiser from data stored.
+     *
+     * @param context the job
+     * @param <V>     the value type of the serialiser
+     * @return the serialiser from this configuration
+     * @throws IOException          if de-serialisation could not happen
+     * @throws NullPointerException if anything is null
+     */
+    public static <V> Serialiser<Object, V> getSerialiser(final JobContext context) throws IOException {
+        Objects.requireNonNull(context, "context");
+        return getSerialiser(context.getConfiguration());
+    }
+
+    /**
+     * Creates the de-serialiser from the information stored in a given configuration. This uses a JSON serialiser to
+     * create the serialiser from data stored.
+     *
+     * @param conf the configuration object for a job
+     * @param <V>  the value type of the serialiser
+     * @return the serialiser from this configuration
+     * @throws IOException          if de-serialisation could not happen
+     * @throws NullPointerException if parameter is null
+     */
+    @SuppressWarnings("unchecked")
+    public static <V> Serialiser<Object, V> getSerialiser(final Configuration conf) throws IOException {
+        Objects.requireNonNull(conf, "conf");
+        String serialConfig = conf.get(PalisadeInputFormat.SERLIALISER_CONFIG_KEY);
```

Comment:
You can remove the 'PalisadeInputFormat.' bit

Reply:
Done